### PR TITLE
Sprint 17 -> dev

### PR DESCRIPTION
**Pull request by mjoyce91**
Sprint 17 work for the dev branch. Does not include Release Candidate-specific changes.

Tweaks :wrench:
- https://github.com/18F/State-TalentMAP/pull/1594 - Shared config for Profile menus
- https://github.com/18F/State-TalentMAP/pull/1640 - Stale token interceptors

Features :star:
- https://github.com/18F/State-TalentMAP/pull/1605 - Position Details redesign / Similar Positions section
- https://github.com/18F/State-TalentMAP/pull/1619 - Favorites Page redesign
- https://github.com/18F/State-TalentMAP/pull/1687 - Search results card design update

### CODE REVIEW

File: src/Components/CheckBox/__snapshots__/CheckBox.test.jsx.snap [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 exports[`CheckBoxComponent matches snapshot 1`] = `
 <div
-  className="usa-grid-full tm-checkbox "
+  className="usa-grid-full  tm-checkbox "
```
Comment (amirbey) [inline src/Components/CheckBox/__snapshots__/CheckBox.test.jsx.snap]: double white space

File: src/Components/CheckBox/__snapshots__/CheckBox.test.jsx.snap [hunk 2]
```diff
@@ -27,7 +27,7 @@ exports[`CheckBoxComponent matches snapshot 1`] = `
 
 exports[`CheckBoxComponent matches snapshot when small = true 1`] = `
 <div
-  className="usa-grid-full tm-checkbox tm-checkbox-small"
+  className="usa-grid-full  tm-checkbox tm-checkbox-small"
```
Comment (amirbey) [inline src/Components/CheckBox/__snapshots__/CheckBox.test.jsx.snap]: double white space

File: src/Components/FavoritePositions/FavoritePositions.jsx [hunk 3]
```diff
@@ -2,25 +2,43 @@ import React from 'react';
...
+        />
+      </div>
+    </div>
     {
       favoritePositionsIsLoading && !favoritePositionsHasErrored &&
         <Spinner type="homepage-position-results" size="big" />
     }
-    <FavoritePositionsList
+    <HomePagePositionsList
+      positions={favorites.results}
```
Comment (amirbey) [inline src/Components/FavoritePositions/FavoritePositions.jsx]: is this positions variable used anywhere? i don't see it used in this file

File: src/Components/FavoritePositions/__snapshots__/FavoritePositions.test.jsx.snap [hunk 4]
```diff
@@ -141,21 +196,107 @@ exports[`FavoritePositionsComponent matches snapshot 1`] = `
...
+            "status": "active",
+            "tour_of_duty": "2 YRS (2 R & R)",
+            "user": "John Doe",
+          },
           "description": Object {
-            "content": "The quick brown fox jumped over the lazy dog.",
-            "date_created": "2018-01-12T15:09:22.952226Z",
+            "content": "The Office Management Specialist (OMS) for the Deputy Chief of Mission (DCM) provides office management services in this Mission of six agencies, 387 American and Locally Employed Staff (LES) positions, and 279 guard force contractors.  The OMS manages Executive Office operations and positionstains the DCM’s calendar in support of overall Mission goals and objectives.  The OMS (DCM) drafts routine correspondence, memoranda and cables as appropriate and ensures documents are cleared and delivered in a timely fashion. The OMS serves as back up to the Ambassador’s OMS.  POCs areSandra McInturff, DCM OMS, McInturffSL@state.gov, 231-77-677-7000, ext. 7452; Samue Watson, DCM, WatsonSR@state.gov, 231-77-677-7000, ext. 7452. Please submit lobbying documents to the 360 Community Lobbying Center http://appcenter.state.sbu/CLC. (Revised 1/2017)
```
Comment (amirbey) [inline src/Components/FavoritePositions/__snapshots__/FavoritePositions.test.jsx.snap]: why is there an empty line at the end of this string?

File: src/Components/HomePagePositionsList/__snapshots__/HomePagePositionsList.test.jsx.snap [hunk 5]
```diff
@@ -730,54 +781,71 @@ exports[`HomePagePositionsList displays two rows 1`] = `
...
+              "status": "active",
+              "tour_of_duty": "2 YRS (2 R & R)",
+              "user": "John Doe",
+            },
             "description": Object {
-              "content": "The quick brown fox jumped over the lazy dog.",
-              "date_created": "2018-01-12T15:09:22.952226Z",
+              "content": "The Office Management Specialist (OMS) for the Deputy Chief of Mission (DCM) provides office management services in this Mission of six agencies, 387 American and Locally Employed Staff (LES) positions, and 279 guard force contractors.  The OMS manages Executive Office operations and positionstains the DCM’s calendar in support of overall Mission goals and objectives.  The OMS (DCM) drafts routine correspondence, memoranda and cables as appropriate and ensures documents are cleared and delivered in a timely fashion. The OMS serves as back up to the Ambassador’s OMS.  POCs areSandra McInturff, DCM OMS, McInturffSL@state.gov, 231-77-677-7000, ext. 7452; Samue Watson, DCM, WatsonSR@state.gov, 231-77-677-7000, ext. 7452. Please submit lobbying documents to the 360 Community Lobbying Center http://appcenter.state.sbu/CLC. (Revised 1/2017)
```
Comment (amirbey) [inline src/Components/HomePagePositionsList/__snapshots__/HomePagePositionsList.test.jsx.snap]: why empty line in string?

File: src/Components/HomePagePositionsSection/HomePagePositionsSection.jsx [hunk 6]
```diff
@@ -19,33 +20,42 @@ const propTypes = {
...
+  const shouldShowAlert = !hasErrored && positions && !positions.length;
+  const shouldShowErrorAlert = hasErrored && !isLoading;
+  const shouldDisplaySpinner = useSpinner && isLoading;
   return (
     <div className="usa-grid-full positions-section">
       <PositionsSectionTitle
         title={
           <span className="positions-section-title">
-            <FontAwesome name={icon} />
+            { !!icon.length && <FontAwesome name={icon} /> }
```
Comment (amirbey) [inline src/Components/HomePagePositionsSection/HomePagePositionsSection.jsx]: double negative ????

File: src/Components/PositionDetailsItem/PositionDetailsContact/PositionDetailsContact.jsx [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+      newWebsiteContent: { value: null },
+      newPocContent: { value: null },
+    };
+  }
+
+  toggleWebsiteEditor() {
+    // reset any alert messages
+    this.props.resetDescriptionEditMessages();
+    const { shouldShowWebsiteEditor } = this.state;
+    shouldShowWebsiteEditor.value = !shouldShowWebsiteEditor.value;
```
Comment (amirbey) [inline src/Components/PositionDetailsItem/PositionDetailsContact/PositionDetailsContact.jsx]: this is confusing ... are you just trying to set this to true? or should there be a default/reset value?

File: src/Components/PositionDetailsItem/PositionDetailsContact/PositionDetailsContact.jsx [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    this.setState({ shouldShowWebsiteEditor });
+  }
+
+  togglePocEditor() {
+    // reset any alert messages
+    this.props.resetDescriptionEditMessages();
+    const { shouldShowPocEditor } = this.state;
+    shouldShowPocEditor.value = !shouldShowPocEditor.value;
```
Comment (amirbey) [inline src/Components/PositionDetailsItem/PositionDetailsContact/PositionDetailsContact.jsx]: this is confusing ... are you just trying to set this to true? or should there be a default/reset value?

File: src/Components/PositionDetailsItem/PositionDetailsContact/__snapshots__/PositionDetailsContact.test.jsx.snap [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+exports[`PositionDetailsContact matches snapshot 1`] = `
+<div
+  className="position-details-contact"
+>
+  <div
+    className="contact-container"
+  >
+    <div
+      className="usa-grid-full contact-section website-section"
+    >
```
Comment (amirbey) [inline src/Components/PositionDetailsItem/PositionDetailsContact/__snapshots__/PositionDetailsContact.test.jsx.snap]: can these divs below be on same line?

File: src/Components/PositionTitle/__snapshots__/PositionTitle.test.jsx.snap [hunk 10]
```diff
@@ -14,79 +14,46 @@ exports[`PositionTitleComponent matches snapshot 1`] = `
...
-            <FontAwesome
-              name="arrow-left"
-            />
-             
-            <a
-              className="back-link"
-              onClick={[Function]}
-              role="link"
-              tabIndex="0"
+          <div
```
Comment (amirbey) [inline src/Components/PositionTitle/__snapshots__/PositionTitle.test.jsx.snap]: can the opening div tags be one line?

File: src/Components/PositionTitleSubDescription/__snapshots__/PositionTitleSubDescription.test.jsx.snap [hunk 11]
```diff
@@ -23,12 +25,14 @@ exports[`PositionTitleSubDescriptionComponent matches snapshot when editor is hi
 
 exports[`PositionTitleSubDescriptionComponent matches snapshot when editor is shown 1`] = `
 <div
-  className="usa-width-one-half position-details-header-body editable-position-field"
+  className="usa-width-one-whole position-details-header-body editable-position-field"
 >
-  <strong>
```
Comment (amirbey) [inline src/Components/PositionTitleSubDescription/__snapshots__/PositionTitleSubDescription.test.jsx.snap]: can this div opening tag be on same line?

File: src/Components/PositionTitleSubDescription/__snapshots__/PositionTitleSubDescription.test.jsx.snap [hunk 12]
```diff
@@ -23,12 +25,14 @@ exports[`PositionTitleSubDescriptionComponent matches snapshot when editor is hi
...
+  <div
```
Comment (amirbey) [inline src/Components/PositionTitleSubDescription/__snapshots__/PositionTitleSubDescription.test.jsx.snap]: can this div opening tag be on same line?

File: src/Components/PositionTitleSubDescription/__snapshots__/PositionTitleSubDescription.test.jsx.snap [hunk 13]
```diff
@@ -23,12 +25,14 @@ exports[`PositionTitleSubDescriptionComponent matches snapshot when editor is hi
...
+    className="editable-position-field-title"
```
Comment (amirbey) [inline src/Components/PositionTitleSubDescription/__snapshots__/PositionTitleSubDescription.test.jsx.snap]: can this div opening tag be on same line?

File: src/Components/PositionDetailsItem/PositionDetailsContact/PositionDetailsContact.jsx (hunk 7 above)
Comment (mjoyce91) [reply src/Components/PositionDetailsItem/PositionDetailsContact/PositionDetailsContact.jsx]: This is to toggle the state to the opposite of its previous value.

File: src/Components/CheckBox/__snapshots__/CheckBox.test.jsx.snap (hunk 2 above)
Comment (mjoyce91) [reply src/Components/CheckBox/__snapshots__/CheckBox.test.jsx.snap]: Nat and I are going to look into a library to clean up these class names. Good catch, thanks!

File: src/Components/HomePagePositionsList/__snapshots__/HomePagePositionsList.test.jsx.snap (hunk 5 above)
Comment (mjoyce91) [reply src/Components/HomePagePositionsList/__snapshots__/HomePagePositionsList.test.jsx.snap]: This is some sample content, similar to what we'd receive from State for capsule descriptions. It looks like in the source there are some line breaks, so this isn't something we're adding in ourselves, it's just how the data is formatted on their end.

File: src/Components/CheckBox/__snapshots__/CheckBox.test.jsx.snap (hunk 2 above)
Comment (mjoyce91) [reply src/Components/CheckBox/__snapshots__/CheckBox.test.jsx.snap]: https://github.com/18F/State-TalentMAP/issues/1721

**Pull request merged**